### DISCUSSION TITLE
PSA: Don't use next in loops for performance

### DISCUSSION THREAD

**Discussion by ccuser44** [Epix-Incorporated/Adonis#451] [General]
It used to be that next would make loops more performant.

However with Luau this is no longer the case and the opposite is true.

Use ipairs() for arrays and pairs() for dictionaries.

Also next is less idiomatic

Basically use ipairs() unless it has custom indexes then use pairs()

For normal tables use ipairs()

**Comment by Sceleratis** [comment]
Yeah, I've known for a while but kept using them out of habit/style preferences but have started moving away from using them.

**Comment by Expertcoderz** [comment]
Closing this as it's been over three years and no modern Luau scripter in their right mind even remembers about `next`.
